src/Datadog.Trace/Util/ProcessHelpers.cs

```diff
@@ -0,0 +1,67 @@
+using System.Diagnostics;
+
+namespace Datadog.Trace.Util
+{
+    internal static class ProcessHelpers
+    {
+        /// <summary>
+        /// Wrapper around <see cref="Process.GetCurrentProcess"/>
+        ///
+        /// On .NET Framework the <see cref="Process"/> class is guarded by a
+        /// LinkDemand for FullTrust, so partial trust callers will throw an exception.
+        /// This exception is thrown when the caller method is being JIT compiled, NOT
+        /// when Process.GetCurrentProcess is called, so this wrapper method allows
+        /// us to catch the exception.
+        /// </summary>
+        /// <returns>Returns the result of <see cref="Process.GetCurrentProcess"/></returns>
+        public static Process GetCurrentProcess()
+        {
+            return Process.GetCurrentProcess();
+        }
+
+        /// <summary>
+        /// Wrapper around <see cref="Process.GetCurrentProcess"/> and <see cref="Process.ProcessName"/>
+        ///
+        /// On .NET Framework the <see cref="Process"/> class is guarded by a
+        /// LinkDemand for FullTrust, so partial trust callers will throw an exception.
+        /// This exception is thrown when the caller method is being JIT compiled, NOT
+        /// when Process.GetCurrentProcess is called, so this wrapper method allows
+        /// us to catch the exception.
+        /// </summary>
+        /// <returns>Returns the name of the current process</returns>
+        public static string GetCurrentProcessName()
+        {
+            return Process.GetCurrentProcess().ProcessName;
```

Comment:
We should probably dispose the `Process` instance in the three methods.

Reply:
I modified the ProcessHelpers calls to dispose the Process object in 02a5a6aca09f19cf524c099aec8d1f58795467a9